src/app/services/user-service/user.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable, Output, EventEmitter } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { User } from 'src/app/models/user';
+import { Batch } from 'src/app/models/batch';
+import { Router } from '@angular/router';
+
+@Injectable({
+  	providedIn: 'root'
+})
+export class UserService {
+	@Output() fireIsLoggedIn: EventEmitter<any> = new EventEmitter<any>();
+
+	url: string = 'http://localhost:8080/users/';
+	user: User = new User();
+	batch: Batch = new Batch();
+
+	constructor(private http: HttpClient, private router: Router) { }
+	
+	getAllUsers() {
+		return this.http.get<User[]>(this.url);
+	}
+
+	getUserById(idParam: number){
+		return this.http.get<User>(this.url+idParam).toPromise();
+	}
+
+	createDriver(userName, firstName, lastName, email, phone, batchNum) {
+
+		this.batch.batchNumber = batchNum;
+
+		this.user.userName = userName;
+		this.user.firstName = firstName;
+		this.user.lastName = lastName;
+		this.user.email = email;
+		this.user.phoneNumber = phone;
+		this.user.batch = this.batch;
```

Comment:
are we using this variable before now? can we just instantiate batch here rather than create a separate instance variable?